ui/app/AppLayouts/Wallet/popups/swap/SwapInputParamsForm.qml

```diff
@@ -9,7 +9,9 @@ QtObject {
     property int selectedAccountIndex: 0
     property int selectedNetworkChainId: -1
     property string fromTokensKey: ""
-    property string fromTokenAmount: ""
+    property real fromTokenAmount: 0
     property string toTokenKey: ""
+    property real toTokenAmount: 0
```

Comment:
Are you sure these should be `real` numbers? I thought we'd settle on passing strings around

Reply:
I thought the issue was when sending a big int to the nim layer and my plan is to convert it to string before its sent to the nim layer.

Reply:
I believe the issue is not about a BigInt; so you'll have a number like `0.21342353453454345455` represented as a float-as-string but when you convert that to QML's `real` it will be truncated as a `double/real` simply doesn't have that precision. Therefore I thought that here, passing around strings would be safer. WDYT @micieslak ?

Reply:
@caybro , Ive moved it back to string but would be great if we could still discuss so  understand better for the future :)

Reply:
Whenever possible we should avoid floats/doubles and represent amounts as big integer (represented as a string) and number of decimal places (int, like 18 for eth). Many numbers cannot be represented exactly in floats/doubles what may lead to subtle inconsistencies. Simple example: 0.07 - https://float.exposed/0x3d8f5c29, https://onecompiler.com/javascript/42fgy4757

Reply:
@micieslak thanks makes sense! @caybro can you check if alls okay for you now?